src/express.re

```diff
@@ -1,43 +1,149 @@
-/*
-  There are a few types to update in this file.
-  We're generic in 'a in most types, but really
-  should specify which JS type should be used.
-
-  An example of how to handle JS fns that take
-  more than one type, where some types are
-  containers and some primitives is the the
-  Response.send binding below.
-
-  sendString | sendJson | sendBuffer | sendArray
-  are all bindings to the .send method based on the
-  input type
-
-  tracked in https://github.com/BuckleTypes/bs-express/issues/6
- */
-module Next = {
+type done_;
+/** abstract type which ensure middleware function must either
+    call the [next] function or one of the [send] function on the 
+    response object. 
+
+    This should be a great argument for OCaml, the type system 
+    prevents silly error which in this case would make the server hang */ 
+
+/** TODO : maybe this should be a more common module like bs-error */
+module Error = {
```

Comment:
In express, this type could be a function argument:
```
app.use("/blah", function(err, req, res, next) {
   // err is of that type
});
```
Additionally it could be thrown as an exception, but I introduced that type for the use case above.